contracts/mapping-token/contracts/v2/Backing.sol

```diff
@@ -0,0 +1,249 @@
+    }
+
+    function changeDailyLimit(address mappingToken, uint amount) public onlyOwner  {
+        _changeDailyLimit(mappingToken, amount);
+    }
+
+    // here add InBoundLane, and remote issuing module must add the corresponding OutBoundLane
+    function addInboundLane(address mappingTokenFactory, address inboundLane) external onlyOwner {
+        uint32 bridgedChainPosition = IMessageVerifier(inboundLane).bridgedChainPosition();
+        uint32 bridgedLanePosition = IMessageVerifier(inboundLane).bridgedLanePosition();
+        bytes32 remoteId = keccak256(abi.encodePacked(bridgedChainPosition, bridgedLanePosition, mappingTokenFactory));
+        inboundLanes[remoteId] = inboundLane;
+        emit NewInBoundLaneAdded(mappingTokenFactory, inboundLane);
+    }
+
+    // here add OutBoundLane, and remote issuing module must add the corresponding InBoundLane
+    function addOutboundLane(address outboundLane) external onlyOwner {
+        uint32 bridgedChainPosition = IMessageVerifier(outboundLane).bridgedChainPosition();
+        uint32 bridgedLanePosition = IMessageVerifier(outboundLane).bridgedLanePosition();
+        bytes32 remoteId = keccak256(abi.encodePacked(bridgedChainPosition, bridgedLanePosition));
+        outboundLanes[remoteId] = outboundLane;
+        emit NewOutBoundLaneAdded(outboundLane);
+    }
+
+    /**
+     * @notice reigister new erc20 token to the bridge. Only owner can do this.
+     * @param bridgedChainPosition the bridged chain position, the mapping token will be created on this target chain
+     * @param bridgedLanePosition the bridged lane positon, this register message will be delived to this lane position
+     * @param mappingTokenFactory the bridged mappingTokenFactory address who will receive this message
+     * @param token the original token address
+     * @param name the name of the original token
+     * @param symbol the symbol of the original token
+     * @param decimals the decimals of the original token
+     */
+    function registerErc20Token(
+        uint32 bridgedChainPosition,
+        uint32 bridgedLanePosition,
+        address mappingTokenFactory,
+        address token,
+        string memory name,
+        string memory symbol,
+        uint8 decimals
+    ) external payable onlyOwner {
+        require(tokens[token][bridgedChainPosition] == address(0), "Backing: token has been registered");
+
+        address outboundLane = getOutBoundLane(bridgedChainPosition, bridgedLanePosition);
+        require(outboundLane != address(0), "cannot find outboundLane to send message");
+        bytes memory newErc20Contract = abi.encodeWithSelector(
+            IMappingTokenFactory.newErc20Contract.selector,
+            address(this),
+            ERC20_TOKEN_TYPE,
+            token,
+            thisChainName,
+            name,
+            symbol,
+            decimals
+        );
+        uint64 nonce = IOutboundLane(outboundLane).send_message(mappingTokenFactory, newErc20Contract);
```

Comment:
App layer should deal with fee when send messages.
If fee < market_fee , revert.
If fee > market_fee , return the remaining.